streams/src/test/java/org/apache/kafka/streams/integration/AppShutdownIntegrationTest.java

```diff
@@ -0,0 +1,140 @@
+        final String appId = "appId_" + testId;
+        final String inputTopic = "input" + testId;
+
+        IntegrationTestUtils.cleanStateBeforeTest(CLUSTER, inputTopic);
+
+
+        final StreamsBuilder builder = new StreamsBuilder();
+
+
+        final List<KeyValue<Object, Object>> processorValueCollector = new ArrayList<>();
+
+        builder.stream(inputTopic).process(() -> new ShutdownProcessor(processorValueCollector), Named.as("process"));
+
+        final Properties properties = mkObjectProperties(
+                mkMap(
+                        mkEntry(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers()),
+                        mkEntry(StreamsConfig.APPLICATION_ID_CONFIG, appId),
+                        mkEntry(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory().getPath()),
+                        mkEntry(StreamsConfig.NUM_STANDBY_REPLICAS_CONFIG, "5"),
+                        mkEntry(StreamsConfig.ACCEPTABLE_RECOVERY_LAG_CONFIG, "6"),
+                        mkEntry(StreamsConfig.MAX_WARMUP_REPLICAS_CONFIG, "7"),
+                        mkEntry(StreamsConfig.PROBING_REBALANCE_INTERVAL_MS_CONFIG, "480000")
+                )
+        );
+
+
+        try (final KafkaStreams kafkaStreams = new KafkaStreams(builder.build(), properties)) {
+            final CountDownLatch latch = new CountDownLatch(1);
+
+            kafkaStreams.start();
+            produceMessages(0L, inputTopic);
+
+            latch.await(10, TimeUnit.SECONDS);
+
+            assertThat(processorValueCollector.size(), equalTo(1));
+        }
+    }
+
+    private void produceMessages(final long timestamp, final String streamOneInput) throws ShutdownRequestedException {
+        IntegrationTestUtils.produceKeyValuesSynchronouslyWithTimestamp(
+                streamOneInput,
+                Arrays.asList(
+                        new KeyValue<>(1, "A"),
+                        new KeyValue<>(2, "B"),
+                        new KeyValue<>(3, "C"),
+                        new KeyValue<>(4, "D"),
+                        new KeyValue<>(5, "E")),
+                TestUtils.producerConfig(
+                        CLUSTER.bootstrapServers(),
+                        IntegerSerializer.class,
+                        StringSerializer.class,
+                        new Properties()),
+                timestamp);
+    }
+}
+
+
+class ShutdownProcessor extends AbstractProcessor<Object, Object> {
+    final List<KeyValue<Object, Object>> valueList;
+
+    ShutdownProcessor(final List<KeyValue<Object, Object>> valueList) {
+        this.valueList = valueList;
+    }
+
+    @Override
+    public void init(final ProcessorContext context) {
+//        throw new ShutdownRequestedException("integration test");
+    }
+
+    @Override
+    public void process(final Object key, final Object value) {
+        valueList.add(new KeyValue<>(key, value));
+//        throw new ShutdownRequestedException("integration test");
```

Comment:
uncomment for test to pass, still will not cause a shutdown until request close is added to the rebalance listener